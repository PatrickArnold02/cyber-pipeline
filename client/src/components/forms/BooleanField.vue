<script setup>
import ToggleSwitch from 'primevue/toggleswitch';

// Props
const props = defineProps({
  // Field name inside of model object and errors
  field: {
    type: String,
    required: true
  },
  // Label text
  label: {
    type: String,
    required: true
  },
  // Boolean to disable editing
  disabled: {
    type: Boolean,
    default: false
  },
  // Validation errors from server
  errors: {
    type: Object,
    default() {
      return {}
    }
  }
})

// V-model of field to be edited
const model = defineModel()
</script>

<template>
  <div class="flex w-full align-items-center">
    <!-- InputText Component -->
    <ToggleSwitch
      :id="field"
      :disabled="disabled"
      :invalid="errors[field] ? true : false"
      v-model="model"
      class=""
      style="min-width: 2.5rem"
    />

    <!-- Label -->
    <label
      :for="field"
      class="ml-2"
      >{{ props.label }}</label
    >

    <!-- Error Text -->
    <small
      :id="field + '-help'"
      class="w-6 text-red-600"
      >{{ errors[field] ? errors[field][0].message : '' }}</small
    >
  </div>
</template>
