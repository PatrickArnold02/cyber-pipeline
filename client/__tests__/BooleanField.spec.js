import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
// Import your component
import BooleanField from '@/components/forms/BooleanField.vue'
// PrimeVue component mock
import ToggleSwitch from 'primevue/toggleswitch'

describe('BooleanField', () => {
  const props = {
    field: 'testField',
    label: 'TestLabel',
    disabled: false,
    errors: {
      testField: [{ message: 'Field is required' }]
    }
  }

  it('should have toggleswitch render correctly', () => {
    const wrapper = mount(BooleanField, {
      props
    });

    const toggleSwitch = wrapper.getComponent(ToggleSwitch);

    expect(toggleSwitch).toBeDefined();
  })

  // it('should have label render correctly', () => {
  //   const wrapper = mount(BooleanField, {
  //     props
  //   });

  //   console.log('wrapper',wrapper.html());

  //   const label = wrapper.get('label[for="TestLabel"]');
  //   expect(label.text()).toBe(props.label);
  // })

  it('disables ToggleSwitch when disabled prop is true', () => {
    const wrapper = mount(BooleanField, {
      props: {
        ...props,
        disabled: true
      }
    });

    const toggleSwitch = wrapper.findComponent(ToggleSwitch);
    expect(toggleSwitch.props('disabled')).toBe(true);
  })

  it('enables InputSwitch when disabled prop is false', () => {
    const wrapper = mount(BooleanField, {
      props: {
        ...props,
        disabled: false
      }
    });

    const toggleSwitch = wrapper.findComponent(ToggleSwitch)
    expect(toggleSwitch.props('disabled')).toBe(false);
  })

  // it('displays error message when errors prop contains error for field', () => {
  //   const wrapper = mount(BooleanField, {
  //     ...props,
  //     error: {testField: ["Error"]}
  //   })

  //   // Check if error message is displayed
  //   const errorText = wrapper.get('small').text()
  //   expect(errorText).toBe(props.errors.testField[0].message)
  // })
})
