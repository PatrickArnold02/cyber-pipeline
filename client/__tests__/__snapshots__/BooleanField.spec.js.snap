// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BooleanField > matches the snapshot 1`] = `
"<div class="flex w-full items-center">
  <!-- InputText Component -->
  <div class="p-toggleswitch p-component p-invalid" style="position: relative; min-width: 2.5rem;" id="testField" data-pc-name="toggleswitch" pc7="" data-pc-section="root" data-p-checked="false" data-p-disabled="false"><input type="checkbox" role="switch" class="p-toggleswitch-input" aria-checked="false" aria-invalid="true" data-pc-section="input">
    <div class="p-toggleswitch-slider" data-pc-section="slider">
      <div class="p-toggleswitch-handle" data-pc-section="handle"></div>
    </div>
  </div><!-- Label --><label for="testField" class="ml-2">TestLabel</label><!-- Error Text --><small id="testField-help" class="w-6/12 text-red-600">Field is required</small>
</div>"
`;
